osu.Game/Graphics/Backgrounds/BeatmapBackgroundWithStoryboard.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Timing;
+using osu.Game.Beatmaps;
+using osu.Game.Storyboards.Drawables;
+
+namespace osu.Game.Graphics.Backgrounds
+{
+    public class BeatmapBackgroundWithStoryboard : BeatmapBackground
+    {
+        public BeatmapBackgroundWithStoryboard(WorkingBeatmap beatmap, string fallbackTextureName = "Backgrounds/bg1")
+            : base(beatmap, fallbackTextureName)
+        {
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            LoadComponentAsync(new DrawableStoryboard(Beatmap.Storyboard)
+                {
+                    Clock = new InterpolatingFramedClock(Beatmap.Track),
+                },
+                loaded =>
+                {
+                    AddInternal(loaded);
+                    if (Beatmap.Storyboard.ReplacesBackground)
+                        Sprite.FadeOut(300, Easing.OutQuint);
```

Comment:
Is this matching how things are displayed during gameplay somehow? Curious why it's doing a fade as opposed to just showing the storyboard/video from the get go.

Reply:
Backgrounds have scaling applied to account for parallax so it can't match gameplay 1:1.
For fading, I just checked player and there's no fading applied. I'll remove that in the next commit.

Reply:
Just keep in mind that with your current implementation, you're loading a background then displaying the storyboard on top. While this may visually work, I'm not sure if it's the best structure for this implementation (as opposed to not deriving from `BeatmapBackground` and handling whether the background or storyboard should be displayed locally, only initialising one or the other).

It's an isolated use case so wouldn't want to spend too much time on it (would probably be okay with how you have it until someone reports performance issues or otherwise), but if you're looking to put in the extra effort it may be worthwhile experimenting with changing the structure to as I propose here and see if it works well or not.